### ISSUE TITLE
Enjoi (joi?) is not strict for boolean

### ISSUE THREAD

**Issue by cmawhorter** [tlivings/enjoi#34]
```js
Enjoi({ type: 'boolean' }).validate(1, console.log) // valid; value === true
Enjoi({ type: 'boolean' }).validate('1', console.log) // valid; value === true
```

These should both be invalid or maybe add strict mode to options to enable it?

**Comment by tlivings** [comment]
These values are coerced to support URL parameters, which are strings.

**Comment by tlivings** [comment]
+1 to strict mode.

**Comment by tlivings** [comment]
So, the above case won't work — that's just coercion occurring from swaggerize like I mentioned. 

However, I am adding support for strictMode because `'true'` does pass when it shouldn't.

**Comment by tlivings** [comment]
Fixed in 3.2.0.
